### ISSUE TITLE
Fix S3966 FP: Warning emitted when iterating through a dictionary to dispose each instance.

### ISSUE THREAD

**Issue by joegoldman2** [SonarSource/sonar-dotnet#8405]
### Description

A warning S3966 is emitted when iterating through a dictionary to dispose each instance. 

### Repro steps

```cs
Dictionary<string, Disposable> disposables = new();

foreach (var (_, disposable) in disposables)
{
    await disposable.DisposeAsync(); // S3966 here
}

List<Disposable> disposables2 = new();

foreach (var disposable in disposables2)
{
    await disposable.DisposeAsync(); // no warning here
}

public class Disposable : IAsyncDisposable
{
    public ValueTask DisposeAsync()
    {
        return new ValueTask();
    }
}
```

### Expected behavior

No warnings emitted.

### Actual behavior

A warning `S3966 Resource 'disposable' has already been disposed explicitly or through a using statement implicitly. Remove the redundant disposal.` is emitted when iterating through a dictionary. There is no warning when iterating through a list (I didn't test all collection types).

### Related information

* C#/VB.NET Plugins version: C# 12
* Visual Studio version: 17.8.1
* MSBuild / dotnet version: .NET 8.0.100
* Operating System Windows: 11 22H2

**Comment by mary-georgiou-sonarsource** [comment]
Hello @joegoldman2,
Thanks for the report.
I confirm this is a false positive issue and it's related to tuples.
We already have issue #7497 regarding this case so I'll close this one.
